inspire_json_merger/pre_filters.py

```diff
@@ -201,3 +205,16 @@ def clean_root_for_acquisition_source(root, head, update):
 
 filter_documents_same_source = partial(keep_only_update_source_in_field, 'documents')
 filter_figures_same_source = partial(keep_only_update_source_in_field, 'figures')
+
+
+def update_material(root, head, update):
+    if "erratum" in get_value(thaw(update), 'dois.material'):
+        return root, head, update
+    for field in FIELDS_WITH_MATERIAL_KEY:
+        updated_field = []
+        for item in update.get(field, []):
+            updated_item = item.set('material', 'erratum')
+            updated_field.append(updated_item)
+        if updated_field:
+            update = update.set(field, v(*updated_field))
```

Comment:
The idiomatic way is to use [`transform`](https://pyrsistent.readthedocs.io/en/latest/api.html#pyrsistent.PMap.transform) here (with `pyrsistent.ny` playing the role of a wildcard):
```suggestion
        update = update.transform([field, ny], lambda element: element.set("material, "erratum"))
```